client/src/components/ui/QRScanner.svelte

```diff
@@ -0,0 +1,110 @@
+<script lang="ts">
+    import QrScanner from 'qr-scanner';
+    import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+    import { Events, ButtonType } from '../types';
+    import { assert } from '../../assert';
+    import { Document, DocumentSchema } from '../../../../model/src/document.ts';
+
+    import { topToastMessage } from '../../pages/dashboard/stores/ToastStore';
+    import Button from './Button.svelte';
+
+    export let maybeId = null as Document['id'] | null;
+
+    const dispatch = createEventDispatcher();
+    let camStart = false as boolean;
+    let qrScanner = null as QrScanner | null;
+
+    async function startCamera() {
+        camStart = true;
+        await qrScanner?.start();
+    }
+
+    function stopCamera() {
+        camStart = false;
+        return qrScanner?.stop();
+    }
+
+    async function handleFileInput() {
+        const uploadElement = document.getElementById('file-selector');
+        assert(uploadElement instanceof HTMLInputElement);
+        assert(uploadElement.files instanceof FileList);
+        const [first] = uploadElement.files;
+        assert(typeof first !== 'undefined');
+        try {
+            const scanData = await QrScanner.scanImage(first, { returnDetailedScanResult: true });
+            if (scanData.data === '') return;
+            maybeId = DocumentSchema.shape.id.parse(scanData.data.trim());
+            dispatch(Events.OnDocumentScan, maybeId);
+        } catch (err) {
+            topToastMessage.enqueue({ title: 'Failed to Find QR code', body: JSON.stringify(err) });
+        }
+        uploadElement.value = '';
+    }
+
+    function setup() {
+        const videoElement = document.getElementById('scan');
```

Comment:
Let's avoid using ID lookups here. Prefer `bind:this`.
```suggestion
        const videoElement = document.getElementById('scan');
```